### PR TITLE
Fix the CI build

### LINKED ISSUE #20: Fix CI Build
Here are the errors: https://github.com/delta-incubator/dat/actions/runs/3312704338

Lint code error:

* Input required and not supplied: token
* Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-nod

Run tests and create tables:

* Process completed with exit code 2.
* Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: actions/checkout, actions/checkout

Not sure why we're getting these errors cause we're not using Node.

### PR DESCRIPTION
**Pull request by MrPowers**
Fixes #20

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,102 +1,72 @@
...
+
+      # Now install _your_ project. This isn't necessary for many types of projects -- particularly
+      # things like Django apps don't need this. But it's a good idea since it fully-exercises the
+      # pyproject.toml and makes that if you add things like console-scripts at some point that
+      # they'll be installed and working.
+      - run: poetry install --no-interaction
+
+      # And finally run tests. I'm using pytest and all my pytest config is in my `pyproject.toml`
+      # so this line is super-simple. But it could be as complex as you need.
+      - run: poetry run pytest
```
Comment (wjones127) [inline .github/workflows/ci.yml:72]:
```suggestion
      - run: make test
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,102 +1,72 @@
...
         with:
-          python-version: ${{ env.PYTHON_VERSION }}
-          cache: 'poetry'          
-
-      - name: Install dependencies and project in dev mode
-        run: |
-          make install
-
-      - name: Lint
-        run: make lint-base lint-mypy
```
Comment (wjones127) [inline .github/workflows/ci.yml:41]: Are we intentionally not linting?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -1,102 +1,72 @@
...
-          make test
-        env:
-          PY_COLORS: 1
-      
-      - name: Reset generated tables
-        if: github.ref == 'refs/heads/main' && github.event_name == 'push'
-        run: make reset-generated-tables
-
-      - name: Write tables
-        run: make write-all
```
Comment (wjones127) [inline .github/workflows/ci.yml:77]: Are we intentionally stopping writing the tables? Is that intentional?

**Review by wjones127 — CHANGES_REQUESTED** [review]
It seems like we should use the makefile in our CI. (unless we want to get rid of the makefile).

File: .github/workflows/ci.yml (hunk 1 above)
Comment (MrPowers) [reply .github/workflows/ci.yml:72]: I think we should consider getting rid of the makefile.  We have Python commands in `main.py` that can be run from the command line via the `click` library.  We're also defining Makefile commands to run the Python commands - it seems circular.  I'm open to changing my mind here, but seems like we're using both `click` and the Makefile to expose a command line interface to interact with this lib.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (MrPowers) [reply .github/workflows/ci.yml:77]: It looks like the code is currently writing the tables, but then not committing or anything, so I'm not really sure what this operation would accomplish.  You'd merge with master, then all the tables would get rewritten, but I don't think anything would be done with the newly generated tables...  I think it's best for the CI to not write the tables so we can try to keep the diffs manageable.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (MrPowers) [reply .github/workflows/ci.yml:41]: OK, so I think I understand what we're doing now.  We're autoformating the code with autopep8 and lining the code with mypy, flake8, and bandit.  The command is currently called [autolinting](https://github.com/delta-incubator/dat/blob/master/Makefile#L10), but think we mean autoformatting.  Do we need to run the code through three different linters?  Perhaps we can autoformat with black and lint with black so we're just using one tool?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (wjones127) [reply .github/workflows/ci.yml:72]: Well those are two different things. The Makefile defines the common CLI invocations used in the repo. It's actually quick useful for newcomers. The includes our scripts, but also the various test and lint scripts (which are unrelated to `click`).

The `click` stuff makes our script invokable by CLI, allowing us to include those invocations in the Makefile, along with all the other commands.

I think that setup makes a lot of sense to me.

Comment (MrPowers) [reply .github/workflows/ci.yml]: @wjones127 - thanks for clarifying.  Made your suggested change.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (MrPowers) [reply .github/workflows/ci.yml:41]: @wjones127 - jfyi, the current CI currently errors out with a ["lint code" error](https://github.com/delta-incubator/dat/issues/20#issue-1420771213), but I'm not sure what's the root cause.  Might be good to get the CI in a working state and then add linting back later.

**Review by wjones127 — APPROVED** [review]
We should get back and add the linting and other tasks back in. But at least we can run the tests for now.

**Pull request merged**